crates/curp-external-api/src/conflict.rs

```diff
@@ -0,0 +1,55 @@
+#![allow(clippy::module_name_repetitions)]
+
+/// Insert into speculative pool
+pub trait SpeculativePoolOp {
+    /// Entry of the pool
+    type Entry;
+
+    /// Inserts a command in to the pool
+    ///
+    /// Returns `Some(Entry)` if a conflict is detected
+    fn insert(&mut self, entry: Self::Entry) -> Option<Self::Entry>;
+
+    /// Returns the number of commands in the pool
+    fn len(&self) -> usize;
+
+    /// Checks if the pool contains some commands that will conflict with all other commands
+    fn is_empty(&self) -> bool;
+
+    /// Removes a command from the pool
+    fn remove(&mut self, entry: Self::Entry);
+
+    /// Returns all commands in the pool
+    fn all(&self) -> Vec<Self::Entry>;
+
+    /// Clears all entries in the pool
+    fn clear(&mut self);
+}
+
+/// Insert into uncommitted pool
+pub trait UncommittedPoolOp {
```

Comment:
There are some operations, such as `clear`, `all`, `len`, and `remove`, that are identical to those in `SpeculativePoolOp`. Shall we extract them into a `CommonPoolOp` trait to avoid duplication?